prepare_data/src/main/groovy/org/orbisgis/osm/FormattingForAbstractModel.groovy

```diff
@@ -0,0 +1,369 @@
+                def inputTable = datasource.getSpatialTable(inputTableName)
+                datasource.execute("    drop table if exists input_road;\n" +
+                        "CREATE TABLE input_road (THE_GEOM GEOMETRY, ID_SOURCE VARCHAR, WIDTH FLOAT, TYPE VARCHAR,\n" +
+                        "SURFACE VARCHAR, SIDEWALK VARCHAR, ZINDEX INTEGER)")
+                inputTable.eachRow { row ->
+                    Float width = getWidth(row.getString("width"))
+                    String type = getAbstractValue(row, mappingForType)
+                    String surface = getAbstractValue(row, mappingForSurface)
+                    String sidewalk = getSidewalk(row.getString("sidewalk"))
+                    Integer zIndex = getZIndex(row.getString("zindex"))
+                    datasource.execute ("insert into input_road values('${row.getGeometry("the_geom")}','${row.getString("id_way")}',${width},'${type}','${surface}','${sidewalk}',${zIndex})".toString())
+                }
+                [outputTableName: "INPUT_ROAD"]
+            }
+    )
+}
+
+/**
+ * This process is used to transform the raw rails table into a table that matches the constraints
+ * of the geoClimate Input Model
+ * @param datasource A connexion to a DB containing the raw rails table
+ * @param inputTableName The name of the raw rails table in the DB
+ * @param mappingForType A map between the target values for the column type in the model
+ *        and the associated key/value tags retrieved from OSM
+ * @return outputTableName The name of the final rails table
+ */
+static IProcess transformRails() {
+    return processFactory.create(
+            "transform the raw rails table into a table that matches the constraints of the GeoClimate Input Model",
+            [datasource          : JdbcDataSource,
+             inputTableName      : String,
+             mappingForType: Map],
+            [outputTableName: String],
+            { datasource, inputTableName, mappingForType ->
+                def inputTable = datasource.getSpatialTable(inputTableName)
+                datasource.execute("    drop table if exists input_rail;\n" +
+                        "CREATE TABLE input_rail (THE_GEOM GEOMETRY, ID_SOURCE VARCHAR, TYPE VARCHAR, ZINDEX INTEGER)")
+                inputTable.eachRow { row ->
+                    String type = getAbstractValue(row, mappingForType)
+                    Integer zIndex = getZIndex(row.getString("layer"))
+
+                    //special treatment if type is subway
+                    if (type == "subway") {
+                        boolean keepSubway
+                        keepSubway = (row.getString("tunnel") != null && row.getString("tunnel") == "no" && row.getString("layer") != null && row.getString("layer").toInt() >= 0) || (row.getString("bridge") != null && (row.getString("bridge") == "yes" || row.getString("bridge") == "viaduct"))
+                        if (!keepSubway) {
```

Comment:
You can put the condition directly in the if statement. Also avoid too long line by using line breaks.

Reply:
Done